lib/matplotlib/legend.py

```diff
@@ -507,10 +512,22 @@ def __init__(self, parent, handles, labels,
         self._loc_used_default = tmp  # ignore changes done by _set_loc
 
         # figure out title fontsize:
-        if title_fontsize is None:
-            title_fontsize = mpl.rcParams['legend.title_fontsize']
-        tprop = FontProperties(size=title_fontsize)
-        self.set_title(title, prop=tprop)
+        if title_prop is None:
+            if title_fontsize is None:
+                title_fontsize = mpl.rcParams["legend.title_fontsize"]
+            self.title_prop = FontProperties(
+                size=title_fontsize)
+        else:
+            self.title_prop = FontProperties._from_any(title_prop)
+            if isinstance(title_prop, dict) and "size" not in title_prop:
+                if title_fontsize is None:
+                    title_fontsize = mpl.rcParams["legend.title_fontsize"]
+                self.title_prop.set_size(title_fontsize)
+            else:
+                if title_fontsize is not None:
+                    self.title_prop.set_size(title_fontsize)
+
+        self.set_title(title, prop=self.title_prop)
```

Comment:
This logic seems convoluted to me and checks the same thing a few different times.  Can it be simplified?  

```suggestion
        self.title_prop = FontProperties._from_any(title_prop)
        if isinstance(title_prop, dict) and "size" not in title_prop:
            if title_fontsize is None:
                title_fontsize = mpl.rcParams["legend.title_fontsize"]
            self.title_prop.set_size(title_fontsize)
        else:
            if title_fontsize is not None:
                self.title_prop.set_size(title_fontsize)

        self.set_title(title, prop=self.title_prop)
```

Reply:
Seems that _from_any can take `None` as its input. Now it looks simpler.